### ISSUE TITLE
feat: allow imports between slices through Public API

### ISSUE THREAD

**Issue by conarti** [conarti/eslint-plugin-feature-sliced#21]
## Problem

Currently, the `layers-slices` rule completely prohibits imports between slices of the same layer:

```js
// If layers: [shared, modules, app]
import { useBooking } from '~/modules/booking'
// ❌ ERROR - import between slices of layer "modules"
```

In architectures where top-level layers contain "modules" (Bounded Contexts), imports between modules through Public API should be allowed.

## Use Case

Architecture with 3 global layers:
```
client/
├─ app/        # Infrastructure
├─ modules/    # Business logic (Bounded Contexts)
└─ shared/     # Shared code
```

Rules:
- Between modules: Only through Public API (`~/modules/[name]`)
- Deep imports forbidden: `~/modules/event/features/...` ❌

## Proposed Solution

Add option to allow imports between slices through their Public API:

```js
featureSliced({
  layers: [
    { name: 'shared', hasSlices: false },
    { name: 'modules', hasSlices: true, allowSliceImports: true },
    { name: 'app', hasSlices: false },
  ],
})
```

Or a global option:

```js
featureSliced({
  layersSlices: {
    allowSliceImportsThroughPublicApi: true,
  },
})
```

## Expected Behavior

```js
import { useBooking } from '~/modules/booking'     // ✅ OK - through Public API
import { useBooking } from '~/modules/booking/features/create' // ❌ ERROR - deep import
```

**Comment by conarti** [comment]
Reopening.

This was closed together with the v2 work, but the option is not in master: `layers-slices` still forbids every import between slices of the same layer. PR #23 implements it. I will retarget it to master and finish it for the 2.1.0 release.
